finality-aleph/src/tcp_network.rs

```diff
@@ -186,25 +168,102 @@ impl TcpAddressingInformation {
             peer_id,
         })
     }
+
+    fn peer_id(&self) -> AuthorityId {
+        self.peer_id.clone()
+    }
+}
+
+/// A representation of TCP addressing information with an associated peer ID, self-signed.
+#[derive(Debug, Hash, Encode, Decode, Clone, PartialEq, Eq)]
+pub struct SignedTcpAddressingInformation {
+    addressing_information: TcpAddressingInformation,
+    signature: Signature,
+}
+
+impl TryFrom<Vec<LegacyTcpMultiaddress>> for SignedTcpAddressingInformation {
+    type Error = AddressingInformationError;
+
+    fn try_from(legacy: Vec<LegacyTcpMultiaddress>) -> Result<Self, Self::Error> {
+        let addressing_information = legacy.try_into()?;
+        // This will never get validated, but that is alright and working as intended.
+        let signature = [
```

Comment:
Maybe:
```
let signature = [0; 64].into();
```

Reply:
omg, thanks for this, I completely forgot this was possible